src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.Aead.cs

```diff
@@ -62,6 +78,15 @@ internal static unsafe void ChaCha20Poly1305Encrypt(
                     CryptographicOperations.ZeroMemory(tag);
                     throw new CryptographicException();
                 }
+
+                GC.KeepAlive(plaintextData);
+                GC.KeepAlive(aadData);
```

Comment:
Why is this needed? I would expect that if `error.Value == null` we don't need plain text anymore.

Reply:
Probably not needed. This should be already pinned due to the `fixed` branch.